### PR TITLE
Docs: Make scripts compatible with Python 3

### PR DESCRIPTION
**Pull request by Simran-B**
https://python-future.org/compatible_idioms.html

- [x] Transform with 2to3 script
- [x] Review automatic transforms
- [x] Optimize code
- [x] Test scripts with Python 2 (did it actually use this version in test run?)
- [x] Test scripts with Python 3 (did it actually use this version in test run?)

### CODE REVIEW

File: utils/generateExamples.py [hunk 1]
```diff
@@ -384,9 +385,9 @@ def analyzeFile(f, filename):
 
 
 def generateSetupFunction():
-    print
-    print "(function () {\n%s}());" % ArangoshSetup
-    print
+    print()
+    print("(function () {\n%s}());" % ArangoshSetup)
+    print()
```
Comment (dothebart) [inline utils/generateExamples.py]:
```suggestion
    print("\n(function () {\n%s}());\n" % ArangoshSetup)
```

**Comment by Simran-B** [comment]
https://jenkins01.arangodb.biz/view/Documentation/job/arangodb-ANY-examples/418/console
Python seems to be 2.7, not sure how to test 3.x on the same machine... change invocations to python3 or via shebang?

**Comment by Simran-B** [comment]
Looks like there is an encoding related error in python3:

https://jenkins01.arangodb.biz/view/Documentation/job/arangodb-ANY-examples/427/console

```
13:32:28 Generating swagger
13:32:28 while parsing '/work/ArangoDB/Documentation/DocuBlocks/Rest/Views/post_api_view_arangosearch.md'
13:32:28 trying to use handler 'skip_code'
13:32:28 
13:32:28 while parsing file: '/work/ArangoDB/Documentation/DocuBlocks/Rest/Views/post_api_view_arangosearch.md' error: 'ascii' codec can't decode byte 0xe2 in position 1237: ordinal not in range(128)
13:32:28 Traceback (most recent call last):
13:32:28   File "/work/ArangoDB/utils/generateSwagger.py", line 1464, in <module>
13:32:28     getOneApi(infile, name + " - " + ', '.join(filenames), fn)
13:32:28   File "/work/ArangoDB/utils/generateSwagger.py", line 1311, in getOneApi
13:32:28     automat.run((infile, ''))
13:32:28   File "/work/ArangoDB/utils/generateSwagger.py", line 402, in run
13:32:28     raise x
13:32:28   File "/work/ArangoDB/utils/generateSwagger.py", line 390, in run
13:32:28     (newState, cargo) = handler(cargo)
13:32:28   File "/work/ArangoDB/utils/generateSwagger.py", line 1274, in skip_code
13:32:28     return comment((fp, last), r)
13:32:28   File "/work/ArangoDB/utils/generateSwagger.py", line 1252, in comment
13:32:28     line = fp.readline()
13:32:28   File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
13:32:28     return codecs.ascii_decode(input, self.errors)[0]
13:32:28 UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 1237: ordinal not in range(128)
13:32:28 
13:32:28 During handling of the above exception, another exception occurred:
13:32:28 
13:32:28 Traceback (most recent call last):
13:32:28   File "/work/ArangoDB/utils/generateSwagger.py", line 1467, in <module>
13:32:28     raise Exception("while parsing file '%s' error: %s" %(thisfn, x))
13:32:28 Exception: while parsing file '/work/ArangoDB/Documentation/DocuBlocks/Rest/Views/post_api_view_arangosearch.md' error: 'ascii' codec can't decode byte 0xe2 in position 1237: ordinal not in range(128)
13:32:28 FAILED!
```

**Comment by Simran-B** [comment]
18:39:12 Automatic merge failed; fix conflicts and then commit the result. ?!

https://jenkins01.arangodb.biz/view/Documentation/job/arangodb-ANY-examples/431/console

**Comment by Simran-B** [comment]
No errors in my test run using https://github.com/arangodb/oskar/tree/python3-compat and https://github.com/arangodb/arangodb/pull/10320/commits/426bca5acece8b6cda63d6258096f7f9ecfc6382

https://jenkins01.arangodb.biz/view/Documentation/job/arangodb-ANY-examples/449/console

https://github.com/arangodb/docs/pull/246

**Comment by Simran-B** [comment]
It looks like generateLicense.py was already broken in both Python 2 and 3?!

Fails with py2 because of `str(response.read(), "utf-8", "ignore")` - looks like the 3 argument variant is py3 only?

Fails with py3 because strings are passed to `html.startswith(…)` but html is a bytes object.

**Comment by Simran-B** [comment]
Failing in Jenkins:
```
[ 
17:41:39   "utils/generateExamples.py", 
17:41:39   "--outputDir", 
17:41:39   "/work/ArangoDB/Documentation/Examples", 
17:41:39   "--outputFile", 
17:41:39   "/work/ArangoDB/arangosh.examples.js", 
17:41:39   "--arangoshSetup", 
17:41:39   "/work/ArangoDB/Documentation/Scripts/setup-arangosh.js", 
17:41:39   "/work/ArangoDB/Documentation/DocuBlocks", 
17:41:39   "/work/ArangoDB/Documentation/Books/Manual", 
17:41:39   "/work/ArangoDB/Documentation/Books/AQL", 
17:41:39   "/work/ArangoDB/Documentation/Books/HTTP", 
17:41:39   "--storageEngine", 
17:41:39   "mmfiles", 
17:41:39   "--storageEngineAgnostic", 
17:41:39   false, 
17:41:39   "--cluster", 
17:41:39   false 
17:41:39 ]
17:41:39 Traceback (most recent call last):
17:41:39   File "utils/generateExamples.py", line 738, in <module>
17:41:39     loopDirectories()
17:41:39   File "utils/generateExamples.py", line 697, in loopDirectories
17:41:39     analyzeFile(f, filename)
17:41:39   File "utils/generateExamples.py", line 281, in analyzeFile
17:41:39     for line in f:
17:41:39   File "/usr/lib/python3.6/encodings/ascii.py", line 26, in decode
17:41:39     return codecs.ascii_decode(input, self.errors)[0]
17:41:39 UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 1237: ordinal not in range(128)
17:41:39 parsing the examples failed - aborting!
17:41:39 { 
17:41:39   "pid" : 83, 
17:41:39   "status" : "TERMINATED", 
17:41:39   "exit" : 1 
17:41:39 }
```

Might have to specify utf-8 encoding on open (io.open for compatibility). Does the newline option matter in generateExamples.py?

**Comment by Simran-B** [comment]
https://jenkins01.arangodb.biz/view/Documentation/job/arangodb-ANY-examples/579/

**Comment by Simran-B** [comment]
@jsteemann Jenkins run succeeded. Good enough to merge IMO. Should we get into trouble again then we can fix that later.

**Pull request merged**